components/HomeChapters/index.tsx

```diff
@@ -1,44 +1,65 @@
-import React, { useState } from "react";
+import React from "react";
 import style from "./style.module.scss";
+import { Chapter } from "utils/types";
 import { getRegion } from "utils/helpers";
 
-const HomeChapters: React.FC = () => {
+interface Props {
+  chapters: Chapter[];
+}
+
+const HomeChapters: React.FC<Props> = ({ chapters }) => {
   const [region, setRegion] = React.useState("unknown");
 
+  // If allowed, get user's coordinates to pinpoint region.
   React.useEffect(() => {
-    if (navigator) {
-      navigator.geolocation.getCurrentPosition((pos: Position) => {
+    if (navigator)
+    {
+      navigator.geolocation.getCurrentPosition((pos: GeolocationPosition) => {
         const [lat, long] = [pos.coords.latitude, pos.coords.longitude];
         setRegion(getRegion(lat, long));
       });
     }
   });
 
+  // Returns "card" element for each chapter in region.
+  const chaptersCards = chapters.filter((chapter) => chapter.region == region).map((chapter) => {
+    // Format chapter name to remove underscores.
+    var name = chapter.name?.replace(/_/g, " ");
+
+    // Get campus' picture if exists. If not, use default color.
+    var picURL, cardStyle;
+    if (chapter.campusPic?.url)
+    {
+      picURL = chapter.campusPic.url
+    } else
+    {
+      cardStyle = { backgroundColor: 'rgba(234,224,241,1.0)' }
+    }
+
+    // Chapter card elements.
+    return (
+      <a href={"chapters/" + chapter.name} className={style.card} key={JSON.stringify(chapter._id)}>
+        <div className={style.cardOverlay} style={cardStyle}>
+          <div className={style.cardText}>{name}</div>
+        </div>
+        <img
+          src={picURL}
+          className={style.cardImage}
+          alt={name}
+        ></img>
+      </a>
+    );
+  });
+  
   return (
     <div className={style.parentContainer}>
       <h2 className={style.title}>Chapters In Your Region</h2>
       <div className={style.cardWrapper}>
-        <a href="chapters" className={style.card}>
-          <div className={style.cardOverlay}>
-            <div className={style.cardText}>University of Tennessee</div>
-          </div>
-          <img
-            src="/utk-placeholder.jpg"
-            className={style.cardImage}
-            alt="Chapter name here"
-          ></img>
-        </a>
-
-        <a href="chapters" className={style.card}>
-          <div className={style.cardOverlay}>
-            <div className={style.cardText}>Rutgers</div>
-          </div>
-          <img
-            src="/rutgers-placeholder.jpg"
-            className={style.cardImage}
-            alt="Chapter name here"
-          ></img>
-        </a>
+        { chapters.filter((chapter) => chapter.region == region).length != 0
+          ? chaptersCards
+          : <h2>We were not able to locate any chapters in your region!</h2>
```

Comment:
Instead of displaying this message when the geo is turned off/not matching a region, it might be better to load two or three random chapters, and instead of saying "chapters in your region" at the top, it would say "our chapters."

Reply:
Thanks for the suggestion! It has been implemented.